notebook/notebookapp.py

```diff
@@ -346,6 +347,45 @@ def start(self):
         self.log.info("Wrote hashed password to %s" % self.config_file)
 
 
+class NbserverStopApp(JupyterApp):
+    version = __version__
+    description="Stop currently running notebook server for a given port"
+    kill_cmd='kill'
+    kill_signal='-3'
+
+    flags = dict(
+        json=({'NbserverStopApp': {'json': True}},
+              "Produce machine-readable JSON output."),
+    )
+
+    json = Bool(True, config=True,
+        help="If True, each line of output will be a JSON object with the "
+                  "details from the server info fpyile.")
+
+    port = Integer(8888, config=True,
+        help="Port of the server to be killed. Default 8888")
+
+
+    def parse_command_line(self, argv=None):
+        super(NbserverStopApp, self).parse_command_line(argv)
+        if self.extra_args:
+            self.port=int(self.extra_args[0])
+
+
+    def start(self):
+        server=next((server for server in list_running_servers(self.runtime_dir) if server.get('port')==self.port),None)
+        if server:
+            subprocess.Popen([self.kill_cmd,self.kill_signal,str(server.get('pid'))],stdout=subprocess.PIPE).communicate()
+        else:
+            ports=[s.get('port') for s in list_running_servers(self.runtime_dir)]
+            if ports:
+                print("There is currently no server running on port {}.".format(self.port))
```

Comment:
It we move this above `if ports:` the output for no running servers will be
```bash
There is currently no server running on port 8888
There are currently no running servers
```

Its seems the 

```bash
There are currently no running servers
```

might be better.  Thoughts?